workers/unity/Packages/com.improbable.gdk.transformsynchronization/MonoBehaviours/TransformSynchronization.cs

```diff
@@ -0,0 +1,191 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using Improbable.Gdk.GameObjectRepresentation;
+using Improbable.Transform;
+using Improbable.Worker.Core;
+using Unity.Entities;
+using UnityEngine;
+
+namespace Improbable.Gdk.TransformSynchronization
+{
+    public class TransformSynchronization : MonoBehaviour
+    {
+        [Require] private TransformInternal.Requirable.Reader transformReader;
+
+        public List<TransformSynchronizationReceiveStrategy> ReceiveStrategies;
+        public List<TransformSynchronizationSendStrategy> SendStrategies;
+
+        public bool SetKinematicWhenNotAuthoritative = true;
+
+        private SpatialOSComponent spatialOSComponent;
+        private bool initialised;
+
+        public uint TickNumber
+        {
+            get
+            {
+                if (enabled == false)
+                {
+                    return 0;
+                }
+
+                var manager = spatialOSComponent.World.GetOrCreateManager<EntityManager>();
+                if (!initialised)
+                {
+                    initialised = manager.HasComponent<TransformToSet>(spatialOSComponent.Entity);
+                    if (!initialised)
+                    {
+                        return 0;
+                    }
+                }
+
+                if (transformReader.Authority != Authority.NotAuthoritative)
+                {
+                    return manager.GetComponentData<TicksSinceLastTransformUpdate>(spatialOSComponent.Entity)
+                        .NumberOfTicks + transformReader.Data.PhysicsTick;
+                }
+
+                return manager.GetComponentData<TransformToSet>(spatialOSComponent.Entity).ApproximateRemoteTick;
```

Comment:
Is there ever a case where these components don't exist when you get to this?

Reply:
If you call it during linking and initial enabling.